src/jarabe/main.py

```diff
@@ -347,6 +347,50 @@ def setup_fonts():
     settings.set_property("gtk-font-name", "%s %f" % (face, size))
 
 
+def setup_proxy():
+    g_mode = Gio.Settings('org.sugarlabs.system.proxy').get_string('mode')
+    if g_mode == 'manual':
+        g_http = Gio.Settings('org.sugarlabs.system.proxy.http')
+        g_http_host = g_http.get_string('host')
+        g_http_port = str(g_http.get_int('port'))
+        if(g_http.get_boolean('use-authentication')):
+            text_to_set = "http://%s:%s@%s:%s/" % (
+                g_http.get_string('authentication-user'),
+                g_http.get_string('authentication-password'),
+                g_http_host,
+                g_http_port)
+            os.environ['http_proxy'] = text_to_set
+        else:
+            text_to_set = "http://%s:%s/" % (
+                g_http_host,
+                g_http_port)
+            os.environ['http_proxy'] = text_to_set
+
+        g_https = Gio.Settings('org.sugarlabs.system.proxy.https')
+        text_to_set = "http://%s:%s/" % (
+            g_https.get_string('host'),
+            str(g_https.get_int('port')))
+        os.environ['https_proxy'] = text_to_set
+
+        g_ftp = Gio.Settings('org.sugarlabs.system.proxy.ftp')
+        text_to_set = "ftp://%s:%s/" % (
+            g_ftp.get_string('host'),
+            str(g_ftp.get_int('port')))
+        os.environ['ftp_proxy'] = text_to_set
+
+        g_socks = Gio.Settings('org.sugarlabs.system.proxy.socks')
+        text_to_set = "socks://%s:%s/" % (
+            g_socks.get_string('host'),
+            str(g_socks.get_int('port')))
+        os.environ['socks_proxy'] = text_to_set
```

Comment:
There is so much copy-paste code through this section.  I'd wager that it doesn't improve the readability, so maybe use a loop?